projects/default/controllers/ros/Ros.cpp

```diff
@@ -439,7 +439,7 @@ void Ros::run(int argc, char **argv) {
       rosgraph_msgs::Clock simulationClock;
       double time = mRobot->getTime();
       simulationClock.clock.sec = (int)time;
-      simulationClock.clock.nsec = (time - simulationClock.clock.sec) * 1.0e+9;
+      simulationClock.clock.nsec = round(time - simulationClock.clock.sec) * 1.0e+9;
```

Comment:
Thank you for this, however, we might lose the millisecond precision, I would rather fix it like this:
```suggestion
      simulationClock.clock.nsec = round(1000 * (time - simulationClock.clock.sec)) * 1.0e+6;  // the round prevents precision issues that can then cause issue when using ros timers
```

Reply:
You are right, just checked my test and I was actually doing
` round((time - simulationClock.clock.sec) * 1.0e+9)` with some additional parentheses. But your suggestions should do the trick as well :)